### ISSUE TITLE
Add possibility to paste JSON Template from clipboard

### ISSUE THREAD

**Issue by hcoohb** [deneb-viz/deneb#255]
First I want to say I love Deneb and thanks for all the effort!

One way maybe to improve the workflow might be to allow pasting directly a JSON template.
Sometime, I just want to copy a visual type and style from one report to another, and having to create/paste/save to a file just to import it only once adds steps that could be avoided by just pasting a new template

Proposed feature could look like:
![image](https://user-images.githubusercontent.com/12975783/179909152-80e6b625-43af-4fa4-8de2-562046f28ced.png)

Thanks

**Comment by hcoohb** [comment]
Awesome,
Thanks!

**Comment by dm-p** [comment]
I'm still doing investigation into the things I'd like to improve for v2, and I've taken a look at some of the feasibility around what we can do here.

Power BI visuals do not have access to the modern Clipboard APIs as these need to be enabled on the iframe sandbox that visuals are hosted in, and MS will not permit any other permission other than `allow-scripts`.

However, there are some ways around this, with some considerations to bear in mind. Power BI Desktop does not permit access to the HTML Drag and Drop API. The Power BI Service does, so my plan is to offer the following functionality as the scope of this feature request for v2:

| Story | Power BI Desktop  | Power BI Service |
|-----------|----------------------|-------------------|
| Click button to import file (current functionality) | ✅ | ✅ |
| Drag and drop valid .json file to button | ❌ | ✅ |
| Copy and paste template text from clipboard (as text) | ✅ | ✅ |
| Copy and paste template from clipboard (as .json file) | ❌ | ✅ |

To summarise, if using Desktop, you will be able to:

* Manually import a file (as you do now)
* Paste a template if copied from text somewhere (like a website)

If editing reports in the Service, you will additionally be able to:

* Drag and drop a file to the import button
* Paste a template if copied to the clipboard as a file on your machine

In all cases, templates will **need to be valid Deneb templates** (i.e. completely validate). 

If you're using JSON from elsewhere (or a partially complete template or a Vega/Vega-Lite spec), the story will be as now, i.e. use the editor. I'm not ruling it out for later, but it won't be in-scope for v2.

**Comment by dm-p** [comment]
We'll work on the wording, but the import button text will be contextual, along the following lines:

* Power BI Service

  ![image](https://user-images.githubusercontent.com/10572054/184579765-5706314f-89d6-40ce-9dd1-11ae7c59e4d7.png)

* Power BI Desktop
     
  ![image](https://user-images.githubusercontent.com/10572054/184580068-0c536688-cdab-481b-a42f-0e00e8b74197.png)

**Comment by dm-p** [comment]
Now that Power BI Desktop is on WebView2, behavior is consistent between there and the service; pasting a valid Deneb template file either from Explorer, or from a text editor, or dragging and dropping a file over the drop zone/import button will work. As such this has been expedited to v1.6 and will be available to preview when the alpha builds start back up. I don't have an ETA for this, but it hopefully won't be too far away.

**Comment by dm-p** [comment]
This is now available to test in the 1.6 alpha build. [Please refer to this blog post](https://coacervo.co/deneb_alpha_1_6) for an overview of the release contents, as well as links to the change log on the documentation site and how you can assist with testing and validating 1.6 if you are able.

It will be incredibly useful if you can confirm that this issue is resolved from your end, or if anything is outstanding to ensure it is completed. If I don't receive any further feedback, I'll close this off when 1.6 gets submitted to AppSource. Thanks once again for raising the issue and for using Deneb :)
